packages/web/src/components/gcds-radios/gcds-radios.tsx

```diff
@@ -0,0 +1,467 @@
+import {
+  Validator,
+  defaultValidator,
+  ValidatorEntry,
+  getValidator,
+  requiredValidator,
+} from '../../validators';
+import i18n from './i18n/i18n';
+
+@Component({
+  tag: 'gcds-radios',
+  styleUrl: 'gcds-radios.css',
+  shadow: { delegatesFocus: true },
+  formAssociated: true,
+})
+export class GcdsRadios {
+  @Element() el: HTMLElement;
+
+  @AttachInternals()
+  internals: ElementInternals;
+
+  private shadowElement?: HTMLInputElement;
+
+  private initialValue?: string;
+
+  private optionObject;
+
+  _validator: Validator<string> = defaultValidator;
+
+  /**
+   * Props
+   */
+
+  /**
+   * Options to render radio buttons
+   */
+  @Prop() options!: string | Array<RadioObject>;
+
+  @Watch('options')
+  validateOptions() {
+    let invalidObject = false;
+    // Assign optionsObject from passed options string/object
+    if (typeof this.options == 'object') {
+      this.optionObject = this.options;
+    } else if (typeof this.options == 'string' && this.options.trim() !== '') {
+      this.optionObject = JSON.parse(this.options);
+    } else {
+      this.optionObject = null;
+    }
+
+    // Validate options has type RadioObject
+    if (this.optionObject) {
+      this.optionObject.map(radio => {
+        if (!isRadioObject(radio)) {
+          invalidObject = true;
+        }
+      });
+    }
+
+    // Assign value if passed options has a checked radio
+    if (this.optionObject && !this.value) {
+      this.optionObject.forEach(radio => {
+        if (radio.checked === 'true' || radio.checked === true) {
+          this.value = radio.value;
+          this.internals.setFormValue(radio.value, 'checked');
+        }
+      });
+    }
+
+    // Log error if no or invalid optionsObject
+    this.errors = handleErrors(
+      this.errors,
+      'options',
+      this.optionObject,
+      invalidObject,
+    );
+  }
+
+  /**
+   * Name attribute for an input element.
+   */
+  @Prop({ reflect: true, mutable: false }) name!: string;
+
+  @Watch('name')
+  validateName() {
+    this.errors = handleErrors(this.errors, 'name', this.name);
+  }
+
+  /**
+   * Name of the form field group.
+   */
+  @Prop({ reflect: true, mutable: false }) legend!: string;
+  @Watch('legend')
+  validateLegend() {
+    this.errors = handleErrors(this.errors, 'legend', this.legend);
+  }
+
+  /**
+   * Specifies if a form field is required or not.
+   */
+  @Prop({ reflect: true, mutable: false }) required: boolean;
+
+  /**
+   * Specifies if a form field is required or not.
+   */
+  @Prop({ reflect: true, mutable: false }) hint: string;
+
+  /**
+   * Error message for invalid radio buttons.
+   */
+  @Prop({ reflect: false, mutable: true }) errorMessage: string;
+  @Watch('errorMessage')
+  validateErrorMessage() {
+    if (this.disabled) {
+      this.errorMessage = '';
+    } else if (!this.hasError && this.errorMessage) {
+      this.hasError = true;
+    } else if (this.errorMessage == '') {
+      this.hasError = false;
+    }
```

Comment:
```suggestion
    } else {
      this.hasError = this.errorMessage ? !this.hasError : false;
    }
```

I think we could summarize both `if else` into the above.